### PR TITLE
feat(weekly business): create weekly business

### LINKED ISSUE #15: Ukens bedrift
NoK ønsker funksjonalitet for ukens bedrift der de kan lage en kø med bedrifter som vil vises på nettsiden

### PR DESCRIPTION
**Pull request by Zenjjim**
## Proposed changes
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

Ukens bedrift kan hentes via /api/v1/weekly_business?this_week

Issue number: closes #15 

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] CHANGELOG.md has been updated. [Guide](https://tihlde.slab.com/posts/changelog-z8hybjom)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Pull request title follows [conventional commits](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) (`type(scope): description`)
- [x] Build (`make PR`) was run locally without failures

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: app/career/views/weekly_business.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            return WeeklyBusiness.objects.filter(
+                year=today().year, week=week_nr(today())
+            )
+        in_future_this_year_filter = Q(year=today().year) & Q(
+            week__gte=week_nr(today())
+        )
+        next_year_filter = Q(year__gt=today().year)
+        return WeeklyBusiness.objects.filter(
+            (in_future_this_year_filter) | next_year_filter
+        ).order_by("year", "week")
```
Comment (olros) [inline app/career/views/weekly_business.py:41]: Bare nysgjerrig: Hva skjer her? Får man bedriftene for dette og neste år?

Comment (Zenjjim) [reply app/career/views/weekly_business.py:41]: Dette er for å garantere at jeg henter ut ting for fremtiden. Så hente ut alle uker i fremtiden dette året og alle de neste årene

File: app/career/apps.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class GroupConfig(AppConfig):
```
Comment (Eirsteir) [inline app/career/apps.py:4]:
```suggestion
class CareerConfig(AppConfig):
```

File: app/career/factories/weekly_business_factory.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import factory
+from factory.django import DjangoModelFactory
+
+from app.career.models import WeeklyBusiness
+from app.util import today
+
+
+class WeeklyBusinessFactory(DjangoModelFactory):
+    """Factory that creates a generic weekly business"""
```
Comment (Eirsteir) [inline app/career/factories/weekly_business_factory.py:11]: You can remove this comment.

File: app/career/models/weekly_business.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    def __str__(self):
+        return f"{self.business_name} - {self.year} - {self.week}"
+
+    def save(self, *args, **kwargs):
+        if self.week < 1 or self.week > 52:
+            raise ValueError("Uke må være mellom 1 og 52")
+        if self.year < today().year:
+            raise ValueError("Ukens bedrift kan ikke opprettes i fortiden")
+        try:
+            if WeeklyBusiness.objects.get(week=self.week, year=self.year).id != self.id:
```
Comment (Eirsteir) [inline app/career/models/weekly_business.py:35]: Is it necessary to have this logic here? Django will raise an exception anyways which might be clearer to use at a higher level. I feel like this logic is a bit out of scope for a save-method on a model. 

It might be more clear to create an exception class `WeeklyBusinessAlreadyExistsException` as well.

What do you think?

File: app/career/tests/test_weekly_business.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
+import pytest
+
+from app.career.factories import WeeklyBusinessFactory
+from app.util import today
+
+
+@pytest.fixture()
+def weekly_business():
```
Comment (Eirsteir) [inline app/career/tests/test_weekly_business.py:8]: I think you can put this in `conftest.py` and still use it in this test?

File: app/career/urls.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+from django.conf.urls import include, url
+from rest_framework import routers
+
+from app.career.views import WeeklyBusinessViewSet
+
+router = routers.DefaultRouter()
+router.register("weekly-business", WeeklyBusinessViewSet, basename="weekly-business")
+
+# Register carrer viewpoints here
```
Comment (Eirsteir) [inline app/career/urls.py:9]: Is this comment necessary?

File: app/career/views/weekly_business.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+            )
+        except ValueError as value_error:
+            return Response(
+                {"detail": str(value_error)}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+    def destroy(self, request, *args, **kwargs):
+        super().destroy(request, *args, **kwargs)
+        return Response(
+            {"detail": _("Ukens bedrift ble slettet")}, status=status.HTTP_200_OK
```
Comment (Eirsteir) [inline app/career/views/weekly_business.py:93]: You can remove the translation underscore since we do not use them

**Review by Eirsteir — CHANGES_REQUESTED** [review]
You impress yet again! This was thorough work and completed fast 🎖️ 

I only have some minor comments about semantics and comments 😉

File: app/career/tests/test_weekly_business.py (hunk 5 above)
Comment (Zenjjim) [reply app/career/tests/test_weekly_business.py:8]: jeg setter uke=5 her, som er grunnen til  at jeg ikke bruker den i `conftest`. Tror du det er en god ide å bruke de istedenfor?

File: app/career/models/weekly_business.py (hunk 4 above)
Comment (Zenjjim) [reply app/career/models/weekly_business.py:35]: this is for the error message only. Makes it easier for giving message for frontend. And since this is a valueerror i dont really see the need for a separate exception

Comment (Eirsteir) [reply app/career/models/weekly_business.py:35]: I don't disagree 😉

File: app/career/tests/test_weekly_business.py (hunk 5 above)
Comment (Eirsteir) [reply app/career/tests/test_weekly_business.py:8]: Litt opp til deg, kanskje lettere enn å jobbe med en tilfeldig uke?

**Review by Eirsteir — APPROVED** [review]
Great work @Zenjjim! 🎉
